src/order/rel_iso.lean

```diff
@@ -13,6 +13,78 @@ universes u v w
 variables {α : Type*} {β : Type*} {γ : Type*}
   {r : α → α → Prop} {s : β → β → Prop} {t : γ → γ → Prop}
 
+/-- A relation homomorphism with respect to a given pair of relations `r` and `s`
+is a function `f : α → β` such that `r a b → s (f a) (f b)`. -/
+@[nolint has_inhabited_instance]
+structure rel_hom {α β : Type*} (r : α → α → Prop) (s : β → β → Prop) :=
+(to_fun : α → β)
+(map_rel' : ∀ {a b}, r a b → s (to_fun a) (to_fun b))
+
+infix ` →r `:25 := rel_hom
+
+namespace rel_hom
+
+instance : has_coe_to_fun (r →r s) := ⟨λ _, α → β, λ o, o.to_fun⟩
+
+theorem map_rel (f : r →r s) : ∀ {a b}, r a b → s (f a) (f b) := f.map_rel'
+
+@[simp] theorem coe_fn_mk (f : α → β) (o) :
+  (@rel_hom.mk _ _ r s f o : α → β) = f := rfl
```

Comment:
```suggestion
@[simp] theorem coe_fn_mk (f : α → β) (o : ∀ ⦃a b⦄, r a b → s (f a) (f b)) :
  (rel_hom.mk f o : α → β) = f := rfl

```

Reply:
You could consider also using `⦃a b⦄` elsewhere (in `map_rel` and `map_rel'`). That would be my inclination, but maybe you want to treat `map_rel f` as an implication between two relations instead of a universally quantified statement.

Reply:
This is another issue of consistency. `embedding.coe_fn_mk` is also defined with an `@`, as are `rel_embedding` and `rel_iso`, and `rel_embedding` and `rel_iso` use `{}` rather than `⦃⦄`. I think that if we change one, we should change all of them, at least as far as `{}` rather than `⦃⦄` in `map_rel`, etc., although it might be equal, and just cleaner notation in the case of `coe_fn_mk`?